core/positions/engine.go

```diff
@@ -612,3 +617,17 @@ func (e *Engine) remove(p *MarketPosition) {
 		}
 	}
 }
+
+// calcVWAP calculates the volume weighted average entry price.
+func calcVWAP(vwap num.Decimal, volume int64, addVolume int64, addPrice *num.Uint) num.Decimal {
+	if volume+addVolume == 0 || addPrice == nil {
+		return num.DecimalZero()
+	}
+
+	volumeDec := num.DecimalFromInt64(volume)
+	addVolumeDec := num.DecimalFromInt64(addVolume)
+	addPriceDec := num.DecimalFromUint(addPrice)
+
+	//	return ((vwap * float64(volume)) + (float64(addPrice) * float64(addVolume))) / (float64(volume) + float64(addVolume))
+	return vwap.Mul(volumeDec).Add(addPriceDec.Mul(addVolumeDec)).Div(volumeDec.Add(addVolumeDec))
```

Comment:
Any chance we could use Uints here instead?

Reply:
maybe lets see how it goes first performance wise and decide later?

Reply:
Alright.

Reply:
no, not alright. I changed my mind again.

Reply:
alright then.

Reply:
not so much because of data type, more because all of this is wrong